exporter/awsemfexporter/config.go

```diff
@@ -66,3 +77,52 @@ type Config struct {
 	// logger is the Logger used for writing error/warning logs
 	logger *zap.Logger
 }
+
+type MetricDescriptor struct {
+	// metricName is the name of the metric
+	metricName string `mapstructure:"metric_name"`
+	// unit defines the overwritten value of metric descriptor `unit`
+	unit string `mapstructure:"unit"`
+	// overwrite set to true means the metric descriptor will be overwritten if it is set before emf exporter reads the metric; false means
+	// the descriptor will only by configured if empty.
+	overwrite bool `mapstructure:"overwrite"`
+}
+
+// Validate filters out invalid metricDeclarations and metricDescriptors
+func (config *Config) Validate() {
+	validDeclarations := []*MetricDeclaration{}
+	for _, declaration := range config.MetricDeclarations {
+		err := declaration.Init(config.logger)
+		if err != nil {
+			config.logger.Warn("Dropped metric declaration.", zap.Error(err))
+		} else {
+			validDeclarations = append(validDeclarations, declaration)
+		}
+	}
+	config.MetricDeclarations = validDeclarations
+
+	validDescriptors := []MetricDescriptor{}
+	for _, descriptor := range config.MetricDescriptors {
+		if descriptor.metricName == "" {
+			continue
+		}
+		if _, ok := EMFSupportedUnits[descriptor.unit]; ok {
+			validDescriptors = append(validDescriptors, descriptor)
+		} else {
+			config.logger.Warn(fmt.Sprintf("Dropped metric descriptor. Error: Unsupported unit type %s.", descriptor.unit))
```

Comment:
Use structured logging here too, with the unit being a field.

Reply:
Changed it to field as zap.String(unit)